### ISSUE TITLE
[BUG] [Updated Info for 1.4.13b] Niamh's Bimbo Liqueur scripts have been broken by recent update

### ISSUE THREAD

**Issue by thequestion1990** [Kitteh6660/Corruption-of-Champions-Mod#1039]
### Overview
	Bimbo champagne scripts are failing to properly reset variables back to their original values when the effect wears off.
	
### Game version
	1.0.2_mod_1.4.13b
	
### Expected behavior
	when the champagne effect ends it undoes all the changes it made.
	
### Actual behavior
	In the plains encounter and in the bazaar it failed to reset hip settings. This seems to be occurring because the script is erroneously applying the bimbo body hip size limit and not accounting for the changes during reversion. The message for hip expansion is popping up right after the Niamh scene.

### Steps to reproduce the issue

encounter niamh in the plains and drink from her. wait a couple of hours. be sure to compare values from before and after, which is easily done with CoCEd.

**Comment by Kitteh6660** [comment]
Seems odd. The effect removal hasn't been touched though.

**Comment by Merik2013** [comment]
This could have been interference from the breast refactor you did an update back. Either way, what seems to be going on is that the champaigne events are calling the wrong scripts, but without knowing the internals to the events myself I can only speculate.

**Comment by thequestion1990** [comment]
This could have been interference from the breast refactor you did an update back. Either way, what seems to be going on is that the champaigne events are calling the wrong scripts, but without knowing the internals to the events myself I can only speculate.

**Comment by Stadler76** [comment]
After looking at the code of `bimboChampagne()` in `Niamh.as`:
- Hips never get reverted, because the old value isn't stored.
- It only affects the first breastrow, but it stores the difference to the `biggestTitSize()`:
```as3
player.changeStatusValue(StatusEffects.BimboChampagne,2,5-player.biggestTitSize());
player.breastRows[0].breastRating = 5;
```
- The stat changes are somewhat stored (in value1), but never revert.

Maybe this should be converted to use the new statuseffects-system by @aimozg?

**Comment by thequestion1990** [comment]
any progress on this?

**Comment by thequestion1990** [comment]
I know this used to work fine. It used to revert everything when the effect wore off. It's definitely fixable.

**Comment by thequestion1990** [comment]
Testing this again, the current issue is that Hip size isn't resetting. This seems to be caused by the champagne erroneously triggering the bimbo body effect where it resets your hip size to a minimum, message and everything. I don't believe this was an enforced effect with the champagne in the unmodded game.

**Comment by Stadler76** [comment]
@thequestion1990 wrote:
> I know this used to work fine. It used to revert everything when the effect wore off.

As you can see in [an older blame](https://github.com/Kitteh6660/Corruption-of-Champions-Mod/blame/f0b24dc0aaa31fbe5ca5b1d2c0238eee8af461e5/classes/classes/Scenes/Places/TelAdre/Niamh.as#L486) this was borked from the start.

> It's definitely fixable.

Since fixing this would require a fifth value at least, this isn't easy to fix and I have to make up a fix, that doesn't break saves, too.

I'll look into it, but I can't promise anything. Quick and dirty fix would be removing the hips-change altogether and work out a solution for the stats change.

**Comment by Stadler76** [comment]
Managed to fix it in e3c171115c06d464564f883b4b9930a790a0fb3e

Wasn't an easy task to fix that old bug.
